### PR TITLE
feat: update config automatically

### PR DESCRIPTION
**Pull request by Pangssu**
# Description

When the contract is upgraded, the value of the contract used by each component should be updated. Therefore, when a contract upgrade event occurs, it reads back the value of that contract and changes the config.

### CODE REVIEW

**Review by 0xHansLee — APPROVED** [review]
One minor comment. LGTM

File: utils/service/watcher/contract.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+					<-time.After(time.Second)
+					upgradedCh <- evt
+					continue
+				}
+				cs.log.Info("config updated")
+			case err := <-sub.Err():
+				cs.log.Error("contract upgrade watch has been canceled", "err", err)
+				return
+			case <-cs.ctx.Done():
+				sub.Unsubscribe()
```
[the remark this answers is not in the document]
Comment (kangsorang) [reply utils/service/watcher/contract.go:64]: how about using defer?
```suggestion
				defer close(upgradedCh)
```

[the remark this answers is not in the document]
Comment (seolaoh) [reply utils/service/watcher/contract.go:64]: Right below this line, there is `return`. So It's same to use 'defer' or not.

File: utils/service/watcher/contract.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+				if err != nil {
+					cs.log.Error("failed to update config", "err", err)
+					<-time.After(time.Second)
+					upgradedCh <- evt
+					continue
+				}
+				cs.log.Info("config updated")
+			case err := <-sub.Err():
+				cs.log.Error("contract upgrade watch has been canceled", "err", err)
+				return
```
[the remark this answers is not in the document]
Comment (Pangssu) [reply utils/service/watcher/contract.go:62]: I removed this condition and changed it to the way we originally used the `resubscribeErr()`.

**Review by kangsorang — APPROVED** [review]
lgtm

**Review by seolaoh — APPROVED** [review]
LGTM

**Pull request merged**